src/Workspaces/Core/Portable/Options/OptionKey.cs

```diff
@@ -42,7 +41,7 @@ public bool Equals(OptionKey other)
 
         public override int GetHashCode()
         {
-            var hash = Option.GetHashCode();
+            var hash = Option?.GetHashCode() ?? 0;
```

Comment:
>Option [](start = 23, length = 6)

Do we use `default(OptionKey)` in a a key somewhere? That seems wrong since that does not represent a valid value of `OptionKey`.

Reply:
We don't seem to be doing it internally in our option providers, but there is an existing unit test validating ToString does not throw for a default OptionKey, which failed when I tried to remove the null check for Option in ToString. This change just makes the null Option check consistent across this type.

Any case, I believe we can correctly update the nullability here once we nullable enable this file. Note I tried to nullable enable this file in this PR, but that was causing large number of cascaded changes, so I decided to revert it for this PR.

Reply:
ok. We can do it later. Re the test - the fix is to delete it ;)